user_tools/src/spark_rapids_tools/cloud/dataproc/dataproccluster.py

```diff
@@ -17,23 +17,35 @@
 """
 
 from typing import ClassVar, Type
+from pydantic import field_validator
 
-from spark_rapids_tools.cloud.cluster import ClientCluster, register_client_cluster, ClusterPropMgr, register_cluster_prop_mgr
+from spark_rapids_tools.cloud.cluster import ClientCluster, register_client_cluster, ClusterPropMgr, \
+    register_cluster_prop_mgr
 from spark_rapids_tools.utils.propmanager import PropValidatorSchemaCamel, PropValidatorSchema
 
 
-class DataprocClusterSchema(PropValidatorSchemaCamel):
+class DataprocClusterSchema(PropValidatorSchemaCamel):  # pylint: disable=missing-class-docstring)
     cluster_name: str
     cluster_uuid: str
     project_id: str
     config: dict
 
+    @field_validator('config')
+    def validate_config(cls, config: dict) -> dict:
+        """
+        Validates the cluster config to ensure it is for GCE instead of GKE.
+        """
+        if 'gceClusterConfig' not in config:
+            raise ValueError("'gceClusterConfig' key is missing in config.")
+        return config
+
 
 class DataprocGkeClusterSchema(PropValidatorSchemaCamel):
     cluster_name: str
     cluster_uuid: str
     project_id: str
     config: dict
+    virtual_cluster_config: dict
```

Comment:
Curious why this is required?

Reply:
This is to distinguish the JSON config schema Dataproc GKE from GCE. Dataproc on GCE will not have this entry but Dataproc on GKE will have this field.

Reply:
Sample schema for Dataproc GKE is present at `user_tools/tests/spark_rapids_tools_ut/resources/cluster/dataproc_gke/cpu-00.yaml`

Reply:
So GCE has `gceClusterConfig` under `config` and GKE has `virtualClusterConfig` by which we are distinguishing the 2. I assume this will throw an error if GCE schema is provided here.

Reply:
Yes, when we need to determine the cluster type from the schema, we compare the yaml file with the `DataprocGke` schema and if it matches, we get an instance of `DataprocGkeClientCluster`.

Method used to match schema:
https://github.com/NVIDIA/spark-rapids-tools/blob/a96ff2e99ee105942ab000faf04453673dad7287/user_tools/src/spark_rapids_tools/utils/propmanager.py#L64-L73

Reply:
Thanks @parthosa for the explanation.